File: angular-playlist/first-app/src/app/dashboard/dashboard.component.ts
import {Component} from '@angular/core';

import {Hero} from '../hero/hero'
import {HeroService} from '../hero/hero.service'

@Component({
    selector: 'my-dashboard',
    templateUrl: './dashboard.component.html',
    styleUrls: ['./dashboard.component.css']
})

export class DashboardComponent {
    heroes: Hero[] = [];

    constructor(private heroService: HeroService) {
    }

    ngOnInit(): void {
        this.heroService.getHeroes().then(heroes => {
            let l = heroes.length;
            l = Math.floor(Math.random() * (l - 4))
            this.heroes = heroes.slice(l, l + 4)
        })
    }
}

File: dart-language/dart-web/dart-ng-heroes/lib/src/dashboard_component.dart
import 'package:angular_router/angular_router.dart';
import 'package:angular/angular.dart';

import 'route_paths.dart';
import 'hero.dart';
import 'hero_service.dart';
import 'hero_search_component.dart';
import 'logger_service.dart';

@Component(
  selector: 'my-dashboard',
  templateUrl: 'dashboard_component.html',
  styleUrls: ['dashboard_component.css'],
  directives: [coreDirectives, HeroSearchComponent, routerDirectives],
)
class DashboardComponent implements OnInit {
  List<Hero> heroes;

  final HeroService _heroService;
  final Logger _logger;

  DashboardComponent(this._heroService, this._logger);

  @override
  void ngOnInit() async {
    heroes = (await _heroService.getAll()).skip(1).take(4).toList();
  }

  String heroUrl(int id) => RoutePaths.hero.toUrl(parameters: { idParam: '$id'});
  void printAllLog() => this._logger.logAll();
}